db-service/test/cqn4sql/keyless.test.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Make sure we issue proper errors in case of path expression along keyless entities
+ */
+'use strict'
+
+const cqn4sql = require('../../lib/cqn4sql')
+const cds = require('@sap/cds')
+const { expect } = cds.test
+describe('keyless entities', () => {
+  let model
+
+  beforeAll(async () => {
+    model = await cds.load(__dirname + '/model/keyless').then(cds.linked)
+  })
+
+  it('no foreign keys for join', () => {
+    const { Books } = model.entities
+    const q = SELECT.from(Books).where(`author[ID = 42].book[ID = 42].author.name LIKE 'King'`)
+    expect(() => cqn4sql(q, model)).to.throw(
+      'Path step “author” of “author[…].book[…].author.name” has no valid foreign keys',
```

Comment:
```suggestion
      'Path step “author” of “author[…].book[…].author.name” has no foreign keys',
```
to make it consistent with the other messages

Reply:
good catch :)